ckanext/datastore_profiler/utils/numericstatistics.py

```
@@ -1,54 +1,74 @@
+
+# Local application specific libraries
+from dataclasses import dataclass
 
 
 @dataclass
 class NumericStatistics:
-    '''
-        This class is a Collection of methods that compute statistics (based on input datatype) as required by OpenData 
-    '''
+    """
+    This class computes statistics on numbers.
+
+    Methods included:
+    - numeric_count
+    """
 
     def numeric_count(self, input):
         """
             Method to compute selected statistics on numerical data
-    
+
         Args:
             input (list): List of Numerical values
                             - input list of ints and/or floats and/or nulls
         Returns:
-            output (dict): Dictionary with min, max, median, mean, null_count, number_counts information
-                            - outputs numeric stats on non-null values (and count of nulls)
+            output (dict): Dictionary with min, max, median, mean, null_count,
+            number_counts information
+                            - outputs numeric stats on non-null values (and
+                            count of nulls)
         """
 
-
-        # make sure all inputs can be converted to float
+        # convert "None" stings to None object
         input = [None if item == "None" else item for item in input]
-        # number == number notation is to check for NaNs, since NaN does not equal itself
-        assert all(isinstance(float(x), float) for x in input if x and x == x), "Input to numeric profiler must be all int, float or null"
-        
-        # remove nulls from working data
-        working_data = [float(number) for number in input if number and number == number and number]
+
+        # Check the inputs are valid. Allowed inputs are float, int, NaN, and None
+
+        try:
+            all(isinstance(float(x), float) for x in input if x and x == x)
+        except ValueError:
+            print("Input to numeric profiler must be all int, float, NaN, or null")
```

Do we want to just print something here?
I think we should raise an error